native/src/components/PinchableBox.tsx

```diff
@@ -0,0 +1,317 @@
+import React, { ReactElement } from 'react'
+import { Animated, Image, ImageErrorEventData, NativeSyntheticEvent, View } from 'react-native'
+import {
+  PanGestureHandler,
+  PinchGestureHandler,
+  PanGestureHandlerGestureEvent,
+  PanGestureHandlerStateChangeEvent,
+  PinchGestureHandlerGestureEvent,
+  PinchGestureHandlerStateChangeEvent,
+  State
+} from 'react-native-gesture-handler'
+
+const USE_NATIVE_DRIVER = true
+const ANIMATION_DURATION = 150
+
+type PropsType = {
+  uri: string
+  /**
+   * @param error Error returned by react native of type any
+   */
+  onError: (error: unknown) => void
+}
+
+type StateType = {
+  interactive: boolean
+  viewDimensions?: { width: number; height: number }
+  imageDimensions?: { width: number; height: number }
+}
+
+export class PinchableBox extends React.Component<PropsType, StateType> {
```

Comment:
```suggestion
class PinchableBox extends React.Component<PropsType, StateType> {
```
since you defined default export

Reply:
Why do you use a class component?

Reply:
@klinzo The Animated API is cleaner to call via a class component in my opinion. It was made for them afaik.